### PR TITLE
event_periodic_callback: add convenience wrapper for periodic callbacks

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

We have `event_periodic` to periodically trigger an event (callback in interrupt context) and we have `event_callback` to execute an event callback by the event thread.

Putting both together always requires some boilerplate and is not easy to discover - let's add a `event_periodic_callback` convenience module that ties it all together nicely.

### Testing procedure

A new test case is provided with `tests/event_periodic_callback`.
Event callbacks are executed in thread context.

### CODE REVIEW

**Review by maribu — APPROVED** [review]
ACK. Code looks good. I trust you have run the test application?

**Comment by benpicco** [comment]
Sure and it will be run by CI too.

File: sys/Makefile.dep [hunk 1]
```diff
@@ -559,6 +559,12 @@ ifneq (,$(filter asymcute,$(USEMODULE)))
   USEMODULE += event_callback
 endif
 
+ifneq (,$(filter event_periodic_callback,$(USEMODULE)))
+  USEMODULE += event_callback
+  USEMODULE += event_periodic
+  USEMODULE += event_thread
```
Comment (miri64) [inline sys/Makefile.dep:565]: What is this module needed for? From the code I don't see any obvious reason for this dependency.

Comment (benpicco) [reply sys/Makefile.dep:565]: Which of those modules do you mean?

Comment (miri64) [reply sys/Makefile.dep:565]: `event_thread`. The module that's in the line I commented on...

File: sys/include/event/periodic_callback.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ */
+
+#ifndef EVENT_PERIODIC_CALLBACK_H
+#define EVENT_PERIODIC_CALLBACK_H
+
+#include "event/callback.h"
+#include "event/periodic.h"
+#include "event/thread.h"
```
Comment (miri64) [inline sys/include/event/periodic_callback.h:30]: Same goes for this include btw. I am not sure what of `event/thread.h` is used here.

File: tests/event_periodic_callback/main.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    puts(ctx);
+}
+
+int main(void)
+{
+    event_periodic_callback_t a, b, c;
+
+    event_periodic_callback_init(&a, ZTIMER_MSEC, EVENT_PRIO_MEDIUM, _event_cb, "event A");
+    event_periodic_callback_init(&b, ZTIMER_MSEC, EVENT_PRIO_MEDIUM, _event_cb, "event B");
+    event_periodic_callback_init(&c, ZTIMER_MSEC, EVENT_PRIO_MEDIUM, _event_cb, "event C");
```
Comment (miri64) [inline tests/event_periodic_callback/main.c:36]: You use the queues of `event_thread` here, but this is application code, so maybe you saw the link error and thought it was a missing dependency?

File: sys/Makefile.dep (hunk 1 above)
Comment (benpicco) [reply sys/Makefile.dep:565]: We need an event thread to execute the callbacks.

Comment (miri64) [reply sys/Makefile.dep:565]: Yes, but that's application code (you provide the queue to the init function). An application might as well just use its own runner thread (which should remain an option anyways IMHO).

**Comment by miri64** [comment]
Mhhh... I darn, forgot to disable auto-merge 😅

**Pull request merged**